crates/ika-core/src/dwallet_mpc/mpc_manager.rs

```diff
@@ -322,12 +322,29 @@ impl DWalletMPCManager {
         Ok(())
     }
 
-    fn wait_for_more_messages_and_retry(&mut self, session_id: ObjectID) {
+    fn wait_for_more_messages_and_retry(&mut self, session_id: ObjectID) -> DwalletMPCResult<()> {
+        let epoch_store = self.epoch_store()?;
         if let Some(session) = self.mpc_sessions.get_mut(&session_id) {
             session.received_more_messages_since_last_retry = false;
             session.attempts_count += 1;
-            session.pending_quorum_for_highest_round_number -= 1;
+            // remove outputs of the previous round, as all validators should have failed
+            self.malicious_handler
+                .report_malicious_actors(&party_ids_to_authority_names(
+                    &session
+                        .serialized_full_messages
+                        .get(&(session.next_round_to_advance - 1))
+                        .unwrap_or(&HashMap::new())
+                        .keys()
+                        .cloned()
+                        .collect::<Vec<PartyID>>(),
+                    &*epoch_store,
```

Comment:
I think this code belongs to #1012, but as long as both are merged I don't care much about the order